### PR TITLE
handle DataSourceConfigurationNotFound

### PR DESCRIPTION
**Pull request by snopoke**
http://manage.dimagi.com/default.asp?225561

Tried a few things and this felt like the least messy.
@orangejenny

### CODE REVIEW

File: corehq/apps/userreports/views.py [hunk 1]
```diff
@@ -616,6 +616,15 @@ def page_context(self):
...
             report_form = self.report_form
         except Exception as e:
+            self.template_name = 'userreports/report_error.html'
+            response = {
+                'report_id': self.existing_report.get_id,
+                'is_static': self.existing_report.is_static,
+                'error_message': '',
+                'details': unicode(e)
+            }
+            recreate_message = _('You can delete and recreate this report using the button below, or ' \
```
Comment (dimagimon) [inline corehq/apps/userreports/views.py]: E502 the backslash is redundant between brackets

**Comment by orangejenny** [comment]
Looks fine, but build failure looks real: `django.core.exceptions.AppRegistryNotReady: The translation infrastructure cannot be initialized before the apps registry is ready. Check that you don't make non-lazy gettext calls at import time.`

File: corehq/apps/userreports/views.py [hunk 2]
```diff
@@ -654,6 +640,33 @@ def page_context(self):
...
+                ).format(reverse(EditConfigReportView.urlname, args=[self.domain, self.existing_report._id])))
+            else:
+                error_message = '{} {}'.format(
+                    error_message_base,
+                    _('You can delete and recreate this report using the button below, or '
+                      'report an issue if you believe you are seeing this page in error.')
+                )
+            response['error_message'] = error_message
+            return response
+        elif isinstance(exception, DataSourceConfigurationNotFoundError):
```
Comment (orangejenny) [inline corehq/apps/userreports/views.py]: I'd expected the order of the if and elif here to be switched - if a manually edited report throws a DataSourceNotFoundError, it'll show the generic "can't use report builder" message, but shouldn't it show the more specific DATA_SOURCE_NOT_FOUND_ERROR_MESSAGE that says exactly what the problem was?

Comment (snopoke) [inline corehq/apps/userreports/views.py]: Yea not sure. I was thinking that it doesn't make much sense to show end users a message that says the datasource has been deleted. The action for them is correct (delete and recreate).

Comment (orangejenny) [inline corehq/apps/userreports/views.py]: Ah, end users don't necessarily know what a data source is. Makes sense.

**Pull request merged**